addons/project/models/project.py

```diff
@@ -228,7 +228,9 @@ def _compute_portal_url(self):
     def map_tasks(self, new_project_id):
         """ copy and map tasks from old to new project """
         tasks = self.env['project.task']
-        for task in self.tasks:
+        # We want to copy archived task, but do not propagate an active_test context key
+        task_ids = self.env['project.task'].with_context(active_test=False).search([('project_id', 'in', self.ids)]).ids
```

Comment:
`self.id` seems better here as it keeps the old semantics (copy from single project `self` to other project `new_project_id`).